src/commands/test/lib/included-files-parser.ts

```diff
@@ -1,12 +1,63 @@
-import { IFileDescriptionJson } from "./test-manifest-reader";
+import { IFileDescriptionJson, ITestCloudManifestJson } from "./test-manifest-reader";
 import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import { validXmlFile } from "./xml-util";
+import { out } from "../../../util/interaction";
+import _ = require("lodash");
 
 const invalidCharactersRegexp = /['"!#$%&+^<=>`|]/;
 
+export async function processIncludedFiles(manifest: ITestCloudManifestJson, include: string[], rootDir: string) {
+  if (!include) {
+    return;
+  }
+
+  const filteredFiles = this.filterIncludedFiles(manifest.files, include);
+  const includedFiles = this.parseIncludedFiles(filteredFiles, rootDir);
+  await this.copyIncludedFiles(manifest, includedFiles, rootDir);
+}
+
+export async function copyIncludedFiles(manifest: ITestCloudManifestJson, includedFiles: IFileDescriptionJson[], rootDir: string) {
+  for (const includedFile of includedFiles) {
+    const copyTarget = path.join(path.dirname(rootDir), includedFile.targetPath);
+    await pfs.cp(includedFile.sourcePath, copyTarget);
+    manifest.files.push(includedFile.targetPath);
+  }
+}
+
 export function parseIncludedFiles(includedFiles: string[], rootDir: string): IFileDescriptionJson[] {
   return includedFiles.map((f) => parseIncludedFile(f, rootDir));
 }
 
+export function filterIncludedFiles(manifestFiles: string[], include: string[]): string[] {
+  if (!include) {
```

Comment:
Problematic files can also be in the manifest, so guess that test should really be on allIncludedFiles. But if allIncludedFiles are empty then the result would be [], so there is no need for the special handling at all.

Reply:
Yes - you're right. I know that the `.dll.config` files have to be manually added using the `--include` flag.

I'm going to leave this for now but perhaps future development could include moving this filter to `allIncludedFiles`. Perhaps we could do this if we ever have need to filter out files that are naturally included in the manifest files? It would require some minor restructuring of the way it's working.